packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -69,6 +77,67 @@ export interface IGarbageCollector {
     ): Promise<IGCStats>;
     /** Called when the latest summary of the system has been refreshed. */
     latestSummaryStateRefreshed(result: RefreshSummaryResult, readAndParseBlob: ReadAndParseBlob): Promise<void>;
+    /** Called when a node is changed. Used to detect and log when an inactive node is changed. */
+    nodeChanged(id: string): void;
+}
+
+/** The garbage collection data of each node in the reference graph. */
+interface IGCNodeData {
+    /** The set of routes to other nodes in the graph. */
+    outboundRoutes: string[];
+    /** If the node is unreferenced, the timestamp of when it was marked unreferenced. */
+    unreferencedTimestampMs?: number;
+}
+
+/**
+ * The garbage collection state of the reference graph. It contains a list of all the nodes in the graph and their
+ * GC data.
+ */
+interface IGCState {
+    gcNodes: { [ id: string ]: IGCNodeData };
```

Comment:
can / should this be something like 
type GcState = Map<string, IGCNodeData>; ?

Reply:
I have tried this. The problem is serializing the map. With an object, we can just call serialize on it. With map, we will have to manually serialize it.

Reply:
This will get serialized and put in summary in the next follow up PR.